File: app/backend/src/main/java/at/htlleonding/model/dto/company/CompanyDetailDTO.java
package at.htlleonding.model.dto.company;

import at.htlleonding.model.dto.AddressDTO;
import at.htlleonding.model.dto.invoice.InvoiceOverviewDTO;

import java.util.List;

public record CompanyDetailDTO(long id, String website, String officeMail, String officePhone, AddressDTO address, List<ContactPersonDTO> contactPersons, List<InvoiceOverviewDTO> invoices) {

    public CompanyDetailDTO(long id, String website, String officeMail, String officePhone, Object o, Object o1, Object o2) {
        this(id, website, officeMail, officePhone, null, null, null);
    }
}

File: app/frontend/web/src/app/model/dto/CompanyDetailDTO.ts
import {Address} from "../Address";
import {ContactPerson} from "../ContactPerson";
import {InvoiceOverview} from "./InvoiceOverviewDTO";

export interface CompanyDetailDTO {
    id: number;
    companyName: string;
    website: string;
    officeMail: string;
    officePhone: string;
    address: Address;
    contactPersons: ContactPerson[];
    invoices: InvoiceOverview[];
}